### PR TITLE
ADIOS Backend: Python 3 support

### PR DESCRIPTION
**Pull request by ax3l**
Adds Python 3 support to the ADIOS backend and enables it in testing again.

Related to #186

### CODE REVIEW

**Comment by ax3l** [comment]
ccing @jreadey @giumas ADIOS backend support enabled for Python 3

**Comment by giumas** [comment]
@jreadey Can you add the `py3` branch to Codacy?

firefox_2018-02-15_08-40-04

File: hdf_compass/adios_model/model.py [hunk 1]
```diff
@@ -19,6 +19,7 @@
 import sys
 import os.path as op
 import posixpath as pp
+import six
```
Comment (giumas) [inline hdf_compass/adios_model/model.py]: The idea is to switch to Python 3 and avoid to increase the code complexity with all the `if`s. Look at the other code in the `py3` branch.

Comment (ax3l) [reply hdf_compass/adios_model/model.py]: So you want Python 3+? Fine with me, sure

**Comment by ax3l** [comment]
@giumas updated to Python 3+ only

**Comment by giumas** [comment]
@ax3l Unless we introduce Py2/3 compatibility for all this branch, there is no advantage in just having the ADIOS plugin Py2/3 compatible!

**Comment by ax3l** [comment]
Py3 only is ok with me, go ahead :)

**Comment by jreadey** [comment]
I've added the py3 branch to Codacy.

**Comment by giumas** [comment]
@jreadey Can you also add HDF Compass to the HDF Group's AppVeyor account?

**Comment by jreadey** [comment]
Sure, here it is: https://ci.appveyor.com/project/jreadey/hdf-compass

**Comment by giumas** [comment]
Does it also trigger the build for the `py3` branch?

**Comment by jreadey** [comment]
No.. I guess it stops with the first failed build.
I've updated to just build py3 for now.

**Pull request merged**